### PR TITLE
[Maintenance]: Update linked-list

### PR DESCRIPTION
**Pull request by BethanyG**
- [x] Updated `linked-list` from problem specifications.
- [x] Decided to require the `delete` action, as well as error conditions.
- [x] Added instruction append for new method and error conditions.
- [x]  Added new `toml` file for test cases
- [x]  Added `additional_tests.json` for error handling cases.
- [x] Added a `JinJa` template for test generation.
- [x] Generated new test file.
- [x] Updated `example.py` for new test cases/methods.

### CODE REVIEW

**Comment by BethanyG** [comment]
@kytrinyx  - Taking you up on your offer.  😄   But feel free to also *not* review this entire thing if you're not into it.

The template for this was a bear (and I was glad @meatball133 could help me make it cleaner), but I think we got to a good place in the end.  Also feeling like the addendum is a bit long, but not sure how to shorten it without omitting important information.

**Review by kytrinyx — APPROVED** [review]
I've been ignoring the `linked-list` exercise in the repositories where I've been updating using a generator—getting the template right is so tricky! I'm stoked that you figured it out :-)

I didn't look at the python code, since I'm no python developer, but I had a few tiny nits on other bits.

Lovely work, both of you ❤️ 💜 ❤️

File: exercises/practice/linked-list/.docs/instructions.apped.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+`delete` will take one argument, which is the vaule to be removed from the linked list.
+If the value appears more than once, only the **first** occurance should be removed.
+
+<br>
+
+## Exception messages
+
+Sometimes it is necessary to [raise an exception][raising]. When you do this, you should always include a **meaningful error message** to indicate what the source of the error is. This makes your code more readable and helps significantly with debugging. For situations where you know that the error source will be a certain type, you can choose to raise one of the [built in error types][error types], but should still include a meaningful message.
+
+This particular exercise requires that you use the [raise statement][raise] to "throw" a `ValueError` when a node value being `delet()`-ed is not found in the linked list.  Additionally, an `IndexError` should be thrown if there are no nodes left to `pop()`. The tests will only pass if you both `raise` these `exceptions` and include messages with them.
```
[the remark this answers is not in the document]
Comment (BethanyG) [reply exercises/practice/linked-list/.docs/instructions.apped.md:24]: Yes.  And thank you for noticing.  😄 😉   This is something I often forget in the "scrum" of adding things, so it is good to get a reminder.

**Comment by kytrinyx** [comment]
> At least I hope I caught all of them!

Story of my life!

**Comment by tony-sappe** [comment]
Hello @BethanyG !

Adding a comment to this PR since this introduced changes which run counter to the original problem statement. While updating the problem statement/instructions would be the simple "fix" there is an argument to be made that the additions increase exercise complexity past a "medium" and should be rolled back to preserve the original exercise.

**First item**
```To keep your implementation simple, the tests will not cover error conditions. Specifically: pop or shift will never be called on an empty list.```
New tests are explicitly testing to ensure a specific exception and exception message is raised. Either the tests needed to be rolled back or the problem statement needs to be updated. 

**Second item**
There is no mention of `delete()` in the problem statement and `__len__()` is now required:
```push (insert value at back);
pop (remove value at back);
shift (remove value at front).
unshift (insert value at front);
```

Adding these functions is increasing the difficulty since the linked list object now needs to traverse through all nodes to provide the expected behavior. This is a departure from the previous exercise implementation which solely utilized the head and tail nodes.

**Comment by tony-sappe** [comment]
> Hello @BethanyG !
> 
> Adding a comment to this PR since this introduced changes which run counter to the original problem statement. While updating the problem statement/instructions would be the simple "fix" there is an argument to be made that the additions increase exercise complexity past a "medium" and should be rolled back to preserve the original exercise.
> 
> **First item** `To keep your implementation simple, the tests will not cover error conditions. Specifically: pop or shift will never be called on an empty list.` New tests are explicitly testing to ensure a specific exception and exception message is raised. Either the tests needed to be rolled back or the problem statement needs to be updated.
> 
> **Second item** There is no mention of `delete()` in the problem statement and `__len__()` is now required:
> 
> ```
> push (insert value at back);
> pop (remove value at back);
> shift (remove value at front).
> unshift (insert value at front);
> ```
> 
> Adding these functions is increasing the difficulty since the linked list object now needs to traverse through all nodes to provide the expected behavior. This is a departure from the previous exercise implementation which solely utilized the head and tail nodes.

**alterpatzer** had the same observation and commented in their Iteration 2 solution:
```
# XXX The tests now expect there to be a delete method, though the problem
# description makes no mention of it.  Also, in direct contradiction to the
# description, they expect IndexError if you try to pop or shift from an
# empty list.
```
https://exercism.org/tracks/python/exercises/linked-list/solutions/alterpatzer

**Comment by BethanyG** [comment]
Hi @tony-sappe,

Currently, we are on pause for community contributions.  You can read more about that [here](https://exercism.org/blog/freeing-our-maintainers).

As a consequence,  adding after-the-fact  comments on PRs is not a great way to go about things.  Commenting on an already closed PR ensures that it is only seen if those involved happen to see the email.  Most of our maintainers get many many emails, so it is easy to miss communications like this.  Instead, you should open a topic on the [forum](https://forum.exercism.org/).

> Either the tests needed to be rolled back or the problem statement needs to be updated.

There was a bug with the instruction append for Python.  That has been fixed, and you can see the additional instructions [here](https://exercism.org/tracks/python/exercises/linked-list).

Other discussions about difficulty and methods should be taken up on the forum.

**Pull request merged**